cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1254,18 +1249,13 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             let src = input_to_reg_mem(ctx, inputs[0]);
             let dst = get_output_reg(ctx, outputs[0]);
 
-            let ext_mode = match (src_ty.bits(), dst_ty.bits()) {
-                (1, 8) | (1, 16) | (1, 32) | (8, 16) | (8, 32) => Some(ExtMode::BL),
-                (1, 64) | (8, 64) => Some(ExtMode::BQ),
-                (16, 32) => Some(ExtMode::WL),
-                (16, 64) => Some(ExtMode::WQ),
-                (32, 64) => Some(ExtMode::LQ),
-                (x, y) if x >= y => None,
-                _ => unreachable!(
-                    "unexpected extension kind from {:?} to {:?}",
-                    src_ty, dst_ty
-                ),
-            };
+            let ext_mode = ExtMode::calculate(src_ty.bits(), dst_ty.bits());
+            assert!(
+                src_ty.bits() < dst_ty.bits() && ext_mode.is_none(),
```

Comment:
Shouldn't this be `src_ty.bits() < dst_ty.bits() || ext_mode.is_none()`?

Reply:
Yeah, looking at it again I think it should actually be: `(src_ty.bits() < dst_ty.bits() && ext_mode.is_some()) || ext_mode.is_none()`. Either we have the right size bits and we get a proper `Some(ExtMode)` or we don't and get a `None`--anything else should be an error.

Reply:
Fixed.